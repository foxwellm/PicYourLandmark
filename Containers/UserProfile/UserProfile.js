import React, { Component } from 'react';
import { StyleSheet, Text, View, TouchableOpacity, Image, ImageBackground } from 'react-native';
import { Icon } from 'react-native-elements';

export class UserProfile extends Component {
  constructor() {
    super()
    this.state = {
      points: 30,
      numLandmarksVisited: 3,
      recentLandmark: 'Coors Field'
    }
  }
  render() {
    const { profilePic, currentUserName } = this.props;
    return (
      <View style={styles.viewContainer}>
        <ImageBackground source={require('../../assets/brandenburggate.jpg')} style={styles.imageBackground}>
          <View style={styles.overlay} />
          {
            currentUserName.length > 0 ? <Text style={styles.bannerText}>WELCOME, {this.props.currentUserName.toUpperCase()}!</Text>
              : <Text style={[styles.bannerText, { fontSize: 17, padding: 5 }]}>Please create an account or log in to track your progress!</Text>
          }
          <Icon style={styles.bannerIcon} color="white" name="id-card" type="font-awesome" size={40} paddingTop={15}/>
        </ImageBackground>
        {
          profilePic.length ?
            <TouchableOpacity style={styles.profilePictureContainer} onPress={() => this.props.takeProfilePic()}>
              <View style={styles.pictureContainer} >
                <Image
                  style={styles.profilePicture}
                  source={{ uri: profilePic }}
                />
              </View>
            </TouchableOpacity>
            : <TouchableOpacity style={styles.profilePictureContainer} onPress={() => this.props.takeProfilePic()}>
              <Icon color="#3c4859" name="portrait" type="badge" size={150} />
            </TouchableOpacity>
        }
        {
          currentUserName.length > 0 ? <View style={styles.userInfo}>
          <View style={styles.achievements}>
            <Icon color="#f44336" name="diamond" type="font-awesome" size={20} />
            <Text style={styles.userInfoText}>{this.state.points} gems</Text>
          </View>
          <View style={styles.achievements}>
            <Icon color="#009688" name="university" type="font-awesome" size={20} />
            <Text style={styles.userInfoText}>{this.state.numLandmarksVisited} landmarks visited</Text>
          </View>
          <View style={styles.achievements}>
            <Icon color="#9c27b0" name="flag-checkered" type="font-awesome" size={20} />
            <Text style={styles.userInfoText}>Recent landmark: {this.state.recentLandmark}</Text>
          </View>
        </View> :
        <View style={styles.userInfo}>
        <View style={styles.achievements}>
          <Icon color="#f44336" name="diamond" type="font-awesome" size={20} />
          <Text style={styles.userInfoText}>0 gems</Text>
        </View>
        <View style={styles.achievements}>
          <Icon color="#009688" name="university" type="font-awesome" size={20} />
          <Text style={styles.userInfoText}>0 landmarks visited</Text>
        </View>
        <View style={styles.achievements}>
          <Icon color="#9c27b0" name="flag-checkered" type="font-awesome" size={20} />
          <Text style={styles.userInfoText}>Recent landmark: n/a</Text>
        </View>
      </View>
        }
      </View>
    )
  }
}

const styles = StyleSheet.create({
  viewContainer: {
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    flex: 3,
  },
  imageBackground: {
    flex: 1.5,
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  overlay: {
    position: 'absolute',
    top: 0,
    width: '100%',
    height: '100%',
    backgroundColor: 'black',
    opacity: 0.4,
  },
  achievements: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    marginLeft: '6%',
  },
  userInfo: {
    flex: 1.8,
    display: 'flex',
    flexDirection: 'column',
    flexWrap: 'wrap'
  },
  userInfoText: {
    fontSize: 20,
    padding: 10,
  },
  profilePictureContainer: {
    flex: 1.2,
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  pictureContainer: {
    width: '65%',
  },
  profilePicture: {
    width: '100%',
    height: '90%',
  },
  bannerText: {
    color: 'white',
    fontSize: 18,
    fontWeight: '600',
  }
});

export default UserProfile;