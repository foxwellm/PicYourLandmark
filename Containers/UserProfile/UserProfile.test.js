import React from 'react';
import { UserProfile } from './UserProfile';
import { shallow } from 'enzyme';
import { TouchableOpacity } from 'react-native';

describe('UserProfile', () => {

  let wrapper;
  let mockProfilePic;
  let mockTakeProfilePic;
  let mockCurrentUserName;
  let mockChangeCurrentPage;
  let mockSetLoggedOutMessage;

  beforeEach(() => {
    mockTakeProfilePic = jest.fn();
    mockChangeCurrentPage = jest.fn();
    mockSetLoggedOutMessage = jest.fn();
    mockProfilePic = 'stringURL';
    mockCurrentUserName = ''
    wrapper = shallow(<UserProfile profilePic={mockProfilePic} takeProfilePic={mockTakeProfilePic} currentUserName={mockCurrentUserName} changeCurrentPage={mockChangeCurrentPage} visitedLocations={[]} setLoggedOutMessage={mockSetLoggedOutMessage} points={0}/>);
  });

  it('should match the snapshot when the user has a profile picture', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when the user does not have a profile picture', () => {
    mockProfilePic = '';
    wrapper = shallow(<UserProfile profilePic={mockProfilePic} takeProfilePic={mockTakeProfilePic} currentUserName={mockCurrentUserName} visitedLocations={[]} points={0} setLoggedOutMessage={mockSetLoggedOutMessage} changeCurrentPage={mockChangeCurrentPage}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when there is a current user with no visited locations', () => {
    mockCurrentUserName = 'tester';
    wrapper = shallow(<UserProfile profilePic={mockProfilePic} takeProfilePic={mockTakeProfilePic} currentUserName={mockCurrentUserName} visitedLocations={[]} points={0} setLoggedOutMessage={mockSetLoggedOutMessage} changeCurrentPage={mockChangeCurrentPage}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when there is a current user with 1 or more visited locations', () => {
    mockCurrentUserName = 'tester';
    wrapper = shallow(<UserProfile profilePic={mockProfilePic} takeProfilePic={mockTakeProfilePic} currentUserName={mockCurrentUserName} visitedLocations={[{ name: 'Eiffel Tower' }]} points={0} setLoggedOutMessage={mockSetLoggedOutMessage} changeCurrentPage={mockChangeCurrentPage}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call takeProfilePic when user already has a profile picture', () => {
    wrapper = shallow(<UserProfile profilePic={mockProfilePic} takeProfilePic={mockTakeProfilePic} currentUserName={mockCurrentUserName} visitedLocations={[{ name: 'Eiffel Tower' }]} points={0} setLoggedOutMessage={mockSetLoggedOutMessage} changeCurrentPage={mockChangeCurrentPage}/>);
    wrapper.find(TouchableOpacity).first().simulate('press');
    expect(mockTakeProfilePic).toHaveBeenCalled();
  });

  it('should call takeProfilePic when user does not have a profile picture', () => {
    wrapper = shallow(<UserProfile profilePic={''} takeProfilePic={mockTakeProfilePic} currentUserName={'hello'} visitedLocations={[]} points={0} setLoggedOutMessage={mockSetLoggedOutMessage} changeCurrentPage={mockChangeCurrentPage}/>);
    wrapper.find(TouchableOpacity).at(0).simulate('press');
    expect(mockTakeProfilePic).toHaveBeenCalled();
  });

  it('should call userNeedLogin when user does not have a profile picture and no currentUserName', () => {
    wrapper = shallow(<UserProfile profilePic={''} takeProfilePic={mockTakeProfilePic} currentUserName={''} visitedLocations={[]} points={0} setLoggedOutMessage={mockSetLoggedOutMessage} changeCurrentPage={mockChangeCurrentPage}/>);
    const spy = jest.spyOn(wrapper.instance(), 'userNeedLogin')
    wrapper.find(TouchableOpacity).first().simulate('press');
    expect(spy).toHaveBeenCalled();
  });
});
