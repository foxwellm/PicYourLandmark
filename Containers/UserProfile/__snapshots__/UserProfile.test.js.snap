// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match the snapshot when the user does not have a profile picture 1`] = `
<Component
  style={
    Object {
      "display": "flex",
      "flex": 3,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <ImageBackground
    source={1}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1.5,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "black",
          "height": "100%",
          "opacity": 0.4,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
    <Component
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "600",
          "paddingBottom": "3%",
        }
      }
    >
      WELCOME, 
      USER 1
      !
    </Component>
    <Themed.Icon
      color="white"
      name="id-card"
      size={40}
      type="font-awesome"
    />
  </ImageBackground>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1.2,
        "justifyContent": "space-around",
      }
    }
  >
    <Themed.Icon
      color="#3c4859"
      name="portrait"
      size={150}
      type="badge"
    />
  </TouchableOpacity>
  <Component
    style={
      Object {
        "display": "flex",
        "flex": 1.8,
        "flexDirection": "column",
        "flexWrap": "wrap",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginLeft": "6%",
        }
      }
    >
      <Themed.Icon
        color="#f44336"
        name="diamond"
        size={20}
        type="font-awesome"
      />
      <Component
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        30
         gems
      </Component>
    </Component>
    <Component
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginLeft": "6%",
        }
      }
    >
      <Themed.Icon
        color="#009688"
        name="university"
        size={20}
        type="font-awesome"
      />
      <Component
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        3
         landmarks visited
      </Component>
    </Component>
    <Component
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginLeft": "6%",
        }
      }
    >
      <Themed.Icon
        color="#9c27b0"
        name="flag-checkered"
        size={20}
        type="font-awesome"
      />
      <Component
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        Recent landmark: 
        Coors Field
      </Component>
    </Component>
  </Component>
</Component>
`;

exports[`UserProfile should match the snapshot when the user has a profile picture 1`] = `
<Component
  style={
    Object {
      "display": "flex",
      "flex": 3,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <ImageBackground
    source={1}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1.5,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "black",
          "height": "100%",
          "opacity": 0.4,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
    <Component
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "600",
          "paddingBottom": "3%",
        }
      }
    >
      WELCOME, 
      USER 1
      !
    </Component>
    <Themed.Icon
      color="white"
      name="id-card"
      size={40}
      type="font-awesome"
    />
  </ImageBackground>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1.2,
        "justifyContent": "space-around",
      }
    }
  >
    <Component
      style={
        Object {
          "width": "65%",
        }
      }
    >
      <Component
        source={
          Object {
            "uri": "stringURL",
          }
        }
        style={
          Object {
            "height": "90%",
            "width": "100%",
          }
        }
      />
    </Component>
  </TouchableOpacity>
  <Component
    style={
      Object {
        "display": "flex",
        "flex": 1.8,
        "flexDirection": "column",
        "flexWrap": "wrap",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginLeft": "6%",
        }
      }
    >
      <Themed.Icon
        color="#f44336"
        name="diamond"
        size={20}
        type="font-awesome"
      />
      <Component
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        30
         gems
      </Component>
    </Component>
    <Component
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginLeft": "6%",
        }
      }
    >
      <Themed.Icon
        color="#009688"
        name="university"
        size={20}
        type="font-awesome"
      />
      <Component
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        3
         landmarks visited
      </Component>
    </Component>
    <Component
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginLeft": "6%",
        }
      }
    >
      <Themed.Icon
        color="#9c27b0"
        name="flag-checkered"
        size={20}
        type="font-awesome"
      />
      <Component
        style={
          Object {
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        Recent landmark: 
        Coors Field
      </Component>
    </Component>
  </Component>
</Component>
`;
